File: 01_intro_ai_assisted_dev/code_python/hello_realtek_ai.py
"""
這個檔案在上課時，會由 ChatGPT/OpenWebUI 根據 hello_realtek_manual.py 改寫而來。
上課前可以先故意留白或只放註解。
"""

def greet(name: str = "") -> str:
    if name == "":
        name = "Engineer"
    return f"Hello, {name} from Realtek!"

def main():
    """
    主函數，負責接收使用者輸入的名字，並呼叫 greet 函數印出問候語。
    """
    name = input("請輸入你的名字：")
    print(greet(name))

if __name__ == "__main__":
    main()

File: 01_intro_ai_assisted_dev/code_perl/hello_realtek_ai.pl
#!/usr/bin/perl
use strict;
use warnings;

sub greet {
    my ($name) = @_;
    $name //= "Engineer";  # 使用預設值 "Engineer" 如果沒有提供名稱
    return "Hello, $name from Realtek!";
}

sub validate_name {
    my ($name) = @_;
    unless (defined $name && ref $name eq '') {
        die "Name must be a string";
    }
    return $name;
}

sub main {
    print "請輸入你的名字：";
    my $name = validate_name(<STDIN>);
    print greet($name), "\n";
}

main;